lib/jekyll.rb

```diff
@@ -160,11 +160,11 @@ def sanitized_path(base_directory, questionable_path)
 
       questionable_path.insert(0, "/") if questionable_path.start_with?("~")
       clean_path = File.expand_path(questionable_path, "/")
-      clean_path.sub!(%r!\A\w:/!, "/")
 
-      if clean_path.start_with?(base_directory.sub(%r!\A\w:/!, "/"))
+      if clean_path.start_with?(base_directory)
         clean_path
       else
+        clean_path.sub!(%r!\A\w:/!, "/")
```

Comment:
Is there a test/scenario that shows where this would be useful; can we just drop the whole `clean_path.sub!(%r!\A\w:/!, "/")` bit? I'm trying to think of a legit case where we receive `questionable_path` with a leading drive letter AND `clean_path.start_with?(base_directory)` returns false AND where `File.join(base_directory, clean_path.sub!(%r!\A\w:/!, "/"))` is expected to do something sensible. It seems like, if `clean_path` contains a drive letter`, it is a fully absolute path by definition. So, doing this substitution and then appending it on to `base_directory` would always result in an invalid path.